src/scripts/main.js

```diff
@@ -1,3 +1,104 @@
 'use strict';
 
-// write code here
+const MIN_ROWS = 2;
+const MAX_ROWS = 10;
+
+const MIN_COLUMNS = 2;
+const MAX_COLUMNS = 10;
+
+const container = document.querySelector('.container');
+const tbody = container.querySelector('tbody');
+
+const buttonAppendRow = container.querySelector('.append-row');
+const buttonRemoveRow = container.querySelector('.remove-row');
+
+const buttonAppendColumn = container.querySelector('.append-column');
+const buttonRemoveColumn = container.querySelector('.remove-column');
+
+let currentNumberOfRows = tbody.children.length;
+let currentNumberOfColumns = tbody.lastElementChild.children.length;
+
+buttonAppendRow.appendRow = () => {
+  if (currentNumberOfRows < MAX_ROWS) {
+    tbody.append(tbody.lastElementChild.cloneNode(true));
+  }
+
+  if (++currentNumberOfRows > MIN_ROWS) {
+    buttonRemoveRow.disabled = '';
+  }
+
+  if (currentNumberOfRows === MAX_ROWS) {
+    buttonAppendRow.disabled = 'disabled';
+  }
+};
```

Comment:
The method name 'appendRow' is being assigned to the buttonAppendRow object, which might cause confusion. Consider using a different method name or attaching the method to a different object to avoid potential conflicts.